### PR TITLE
[FLINK-32287][docs] Add doc for truncate table statement

### PR DESCRIPTION
**Pull request by ruanhang1993**
## What is the purpose of the change

Add doc for truncate table statement

## Brief change log

Add doc for truncate table statement

## Verifying this change

This change is a docs work without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by ruanhang1993** [comment]
@luoyuxia Please take a look at this.

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<a name="truncate-statements"></a>
+
+# TRUNCATE TABLE 语句
+
+DESCRIBE 语句用于删除表中的全部数据，但不会删除表本身。TRUNCATE TABLE 语句目前仅支持使用在Batch模式。
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:31]:
```suggestion
TRUNCATE TABLE 语句用于删除表中的全部数据，但不会删除表本身。TRUNCATE TABLE 语句目前仅支持在Batch 模式下使用。
```

Comment (luoyuxia) [reply docs/content.zh/docs/dev/table/sql/truncate.md:31]: Please add the attention just like [`DELETE` statement ](https://nightlies.apache.org/flink/flink-docs-master/zh/docs/dev/table/sql/delete/) in FLINK.
```
注意 目前, DELETE 语句仅支持批模式, 并且要求目标表实现了 [SupportsRowLevelDelete ](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsRowLevelDelete.java)接口。 如果在一个没有实现该接口的表上执行 DELETE，则会抛异常。
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# TRUNCATE TABLE Statement
+
+TRUNCATE TABLE statement are used to delete all rows from a table without dropping the table itself.
+TRUNCATE TABLE statement is only supported in batch execution mode now.
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:30]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# TRUNCATE TABLE Statement
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:27]: Change to `TRUNCATE Statements` just like [`ANALYZE Statements`](http://localhost:1313/docs/dev/table/sql/analyze/)?

File: docs/content/docs/dev/table/sql/truncate.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        " `user` BIGINT NOT NULl comment 'this is primary key'," +
+        " product VARCHAR(32)," +
+        " amount INT," +
+        " ts TIMESTAMP(3) comment 'notice: watermark'," +
+        " ptime AS PROCTIME() comment 'this is a computed column'," +
+        " PRIMARY KEY(`user`) NOT ENFORCED," +
+        " WATERMARK FOR ts AS ts - INTERVAL '1' SECONDS" +
+        ") with (...)");
+
+// print the schema
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:67]: why `// print the schema`?

File: docs/content/docs/dev/table/sql/truncate.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+# TRUNCATE TABLE Statement
+
+TRUNCATE TABLE statement are used to delete all rows from a table without dropping the table itself.
+TRUNCATE TABLE statement is only supported in batch execution mode now.
+
+
+## Run a TRUNCATE TABLE statement
+
+{{< tabs "truncate" >}}
+{{< tab "Java" >}}
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:36]: Why miss `Scala`/`Python` example?

File: docs/content/docs/dev/table/sql/truncate.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+TRUNCATE TABLE statement can be executed with the `executeSql()` method of the `TableEnvironment`. The `executeSql()` method returns nothing for a successful TRUNCATE TABLE operation, otherwise will throw an exception.
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:37]: Actually, it doesn't return nothing. It returns `TABLE_RESULT_OK`. If we can't describe what is returned precisely , I'd like to not to mention it.

File: docs/content/docs/dev/table/sql/truncate.md (hunk 4 above)
Comment (luoyuxia) [reply docs/content/docs/dev/table/sql/truncate.md:67]: Also, please follow the examples in `delete statement` to make sure we can really delete all rows using truncate table statement.

**Review by luoyuxia — COMMENTED** [review]
@ruanhang1993 Thanks for the pr. I left some comments. PTAL

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+under the License.
+-->
+
+<a name="truncate-statements"></a>
+
+# TRUNCATE 语句
+
+TRUNCATE 语句用于删除表中的全部数据，但不会删除表本身。
+<span class="label label-danger">注意</span> 目前, `TRUNCATE` 语句仅支持批模式, 并且要求目标表实现了 {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}} 接口。
+如果在一个没有实现该接口的表上执行 `TRUNCATE`，则会抛异常。目前 Flink 内置的连接器还没有实现该接口。
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:33]: I'd like not to mention `目前 Flink 内置的连接器还没有实现该接口。` If some internal connectors implement it, we may well forget to update this doc. WDYT?

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+Flink SQL> SELECT * FROM Orders;
+    user                         product      amount
+    Lili                           Apple           1
+ Jessica                          Banana           2
+Mr.White                         Chicken           3
+
+Flink SQL> TRUNCATE TABLE Orders;
+// Empty set
+```
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:151]: Please don't forget to select the table again then. It shoud be an empty set.

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+Job ID: bd2c46a7b2769d5c559abd73ecde82e9
+
+Flink SQL> SELECT * FROM Orders;
+    user                         product      amount
+    Lili                           Apple           1
+ Jessica                          Banana           2
+Mr.White                         Chicken           3
+
+Flink SQL> TRUNCATE TABLE Orders;
+// Empty set
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:150]:
```suggestion
[INFO] Execute statement succeed.
```

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+<a name="truncate-statements"></a>
+
+# TRUNCATE 语句
+
+TRUNCATE 语句用于删除表中的全部数据，但不会删除表本身。
+<span class="label label-danger">注意</span> 目前, `TRUNCATE` 语句仅支持批模式, 并且要求目标表实现了 {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}} 接口。
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:32]: Please break a line in here. 
```suggestion

<span class="label label-danger">注意</span> 目前, `TRUNCATE` 语句仅支持批模式, 并且要求目标表实现了 {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}} 接口。
```

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+如果在一个没有实现该接口的表上执行 `TRUNCATE`，则会抛异常。目前 Flink 内置的连接器还没有实现该接口。
+
+<a name="run-a-truncate-statement"></a>
+
+## 执行 TRUNCATE 语句
+
+{{< tabs "truncate" >}}
+{{< tab "Java" >}}
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:40]: Why chinese doc miss `Scala/Python` in tab title?
image

**Review by luoyuxia — COMMENTED** [review]
@ruanhang1993 Thanks for updating. I left minor comments. PTAL.

File: docs/content/docs/dev/table/sql/truncate.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# TRUNCATE Statements
+
+TRUNCATE statements are used to delete all rows from a table without dropping the table itself.
+
+<span class="label label-danger">Attention</span> Currently, `TRUNCATE` statement only supports in batch mode, and it requires the target table connector implements the  {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}}
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:31]: nit: 
```suggestion
<span class="label label-danger">Attention</span> Currently, `TRUNCATE` statement only is supported in batch mode, and it requires the target table connector implements the  {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}}
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+interface to support the row-level delete. An exception will be thrown if trying to `TRUNCATE` the table which has not implements the related interface.
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:32]:
```suggestion
interface to support the row-level delete. An exception will be thrown if trying to `TRUNCATE` a table which has not implemented the related interface.
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+```
+{{< /tab >}}
+{{< tab "Scala" >}}
+```scala
+val env = StreamExecutionEnvironment.getExecutionEnvironment()
+val settings = EnvironmentSettings.newInstance().inBatchMode().build()
+val tEnv = StreamTableEnvironment.create(env, settings)
+
+// register a table named "Orders"
+tEnv.executeSql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)");
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:99]:
```suggestion
tEnv.executeSql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)")
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+// insert values
+tEnv.executeSql("INSERT INTO Orders VALUES ('Lili', 'Apple', 1), ('Jessica', 'Banana', 2), ('Mr.White', 'Chicken', 3)").await();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:101]:
```suggestion
tEnv.executeSql("INSERT INTO Orders VALUES ('Lili', 'Apple', 1), ('Jessica', 'Banana', 2), ('Mr.White', 'Chicken', 3)").await()
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+tEnv.executeSql("SELECT * FROM Orders").print();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:102]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+// +--------------------------------+--------------------------------+-------------+
+// |                           user |                        product |      amount |
+// +--------------------------------+--------------------------------+-------------+
+// |                           Lili |                          Apple |           1 |
+// |                        Jessica |                         Banana |           2 |
+// |                       Mr.White |                        Chicken |           3 |
+// +--------------------------------+--------------------------------+-------------+
+// 3 rows in set
+// truncate the table "Orders"
+tEnv.executeSql("TRUNCATE TABLE Orders").await();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:112]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+tEnv.executeSql("SELECT * FROM Orders").print();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:113]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+// Empty set
+```
+{{< /tab >}}
+{{< tab "Python" >}}
+```python
+env_settings = EnvironmentSettings.in_batch_mode()
+table_env = TableEnvironment.create(env_settings)
+
+# register a table named "Orders"
+table_env.executeSql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)");
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:123]:
```suggestion
table_env.execute_sql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)")
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+# insert values
+table_env.executeSql("INSERT INTO Orders VALUES ('Lili', 'Apple', 1), ('Jessica', 'Banana', 2), ('Mr.White', 'Chicken', 3)").wait();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:125]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 21]
```diff
@@ -0,0 +1,175 @@
...
+table_env.executeSql("SELECT * FROM Orders").print();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:126]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+# +--------------------------------+--------------------------------+-------------+
+# |                           user |                        product |      amount |
+# +--------------------------------+--------------------------------+-------------+
+# |                           Lili |                          Apple |           1 |
+# |                        Jessica |                         Banana |           2 |
+# |                       Mr.White |                        Chicken |           3 |
+# +--------------------------------+--------------------------------+-------------+
+# 3 rows in set
+# truncate the table "Orders"
+table_env.executeSql("TRUNCATE TABLE Orders").wait();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:136]: dito

File: docs/content/docs/dev/table/sql/truncate.md [hunk 23]
```diff
@@ -0,0 +1,175 @@
...
+table_env.executeSql("SELECT * FROM Orders").print();
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:137]: dito

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 24]
```diff
@@ -0,0 +1,181 @@
...
+-->
+
+<a name="truncate-statements"></a>
+
+# TRUNCATE 语句
+
+TRUNCATE 语句用于删除表中的全部数据，但不会删除表本身。
+
+<span class="label label-danger">注意</span> 目前, `TRUNCATE` 语句仅支持批模式, 并且要求目标表实现了 {{< gh_link file="flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTruncate.java" name="SupportsTruncate" >}} 接口。
+如果在一个没有实现该接口的表上执行 `TRUNCATE`，则会抛异常。
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:34]:
```suggestion
如果在一个没有实现该接口的表上执行 `TRUNCATE` 语句，则会抛异常。
```

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 25]
```diff
@@ -0,0 +1,181 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<a name="truncate-statements"></a>
+
+# TRUNCATE 语句
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:29]: The comments in en doc is also suitable for the zh doc.

**Review by luoyuxia — COMMENTED** [review]
@ruanhang1993 Thanks for updating. I left some comments. Should be ready to be merged in next itertation.

File: docs/content/docs/dev/table/sql/truncate.md [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+// Empty set
+```
+{{< /tab >}}
+{{< tab "Python" >}}
+```python
+env_settings = EnvironmentSettings.in_batch_mode()
+table_env = TableEnvironment.create(env_settings)
+
+# register a table named "Orders"
+table_env.executeSql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)")
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:123]:
```suggestion
table_env.execute_sql("CREATE TABLE Orders (`user` STRING, product STRING, amount INT) WITH (...)")
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 27]
```diff
@@ -0,0 +1,175 @@
...
+# insert values
+table_env.executeSql("INSERT INTO Orders VALUES ('Lili', 'Apple', 1), ('Jessica', 'Banana', 2), ('Mr.White', 'Chicken', 3)").wait()
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:125]:
```suggestion
table_env.execute_sql("INSERT INTO Orders VALUES ('Lili', 'Apple', 1), ('Jessica', 'Banana', 2), ('Mr.White', 'Chicken', 3)").wait()
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 28]
```diff
@@ -0,0 +1,175 @@
...
+# +--------------------------------+--------------------------------+-------------+
+# |                           user |                        product |      amount |
+# +--------------------------------+--------------------------------+-------------+
+# |                           Lili |                          Apple |           1 |
+# |                        Jessica |                         Banana |           2 |
+# |                       Mr.White |                        Chicken |           3 |
+# +--------------------------------+--------------------------------+-------------+
+# 3 rows in set
+# truncate the table "Orders"
+table_env.executeSql("TRUNCATE TABLE Orders").wait()
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:136]:
```suggestion
table_env.execute_sql("TRUNCATE TABLE Orders").wait()
```

File: docs/content/docs/dev/table/sql/truncate.md [hunk 29]
```diff
@@ -0,0 +1,175 @@
...
+table_env.executeSql("SELECT * FROM Orders").print()
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:137]:
```suggestion
table_env.execute_sql("SELECT * FROM Orders").print()
```

**Review by luoyuxia — COMMENTED** [review]
@ruanhang1993 Thanks for update. I left some comments. PTAL.

File: docs/content.zh/docs/dev/table/sql/truncate.md [hunk 30]
```diff
@@ -0,0 +1,181 @@
...
+
```
Comment (luoyuxia) [inline docs/content.zh/docs/dev/table/sql/truncate.md:30]: nit:
{{< label Batch >}}

File: docs/content/docs/dev/table/sql/truncate.md [hunk 31]
```diff
@@ -0,0 +1,175 @@
...
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# TRUNCATE Statements
+
```
Comment (luoyuxia) [inline docs/content/docs/dev/table/sql/truncate.md:28]:
```suggestion
{{< label Batch >}}
```
Don't forget to break a new line, you can follow other docs.

**Review by luoyuxia — COMMENTED** [review]
@ruanhang1993 Thanks for update. I left minor comments.

**Review by luoyuxia — APPROVED** [review]
LGTM

**Pull request merged**